### PR TITLE
Guilds UI

### PR DESCRIPTION
**Pull request by Vladislav94**

### CODE REVIEW

File: src/pages/Guilds/routes/GuildsPreview.js [hunk 1]
```diff
@@ -2,50 +2,53 @@ import React, { useContext, useEffect } from 'react';
...
-                    <div className={classes.guildItem} key={index}>
-                        <button
-                            className={classes.guildButton}
-                            disabled={!guild.members?.length}
-                            onClick={ handleClick(guild) }
-                        >
-                            <div className={classes.guildLogo}>
-                                <GuildLogo logo={guild.logo} className={classes.guildLogoImage} />
-                            </div>
+    // const setNumber = amount => {
```
Comment (dmutrokot) [inline src/pages/Guilds/routes/GuildsPreview.js:28]: please leave TODO

**Review by dmutrokot — CHANGES_REQUESTED** [review]

File: src/pages/Guilds/GuildsContext.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            if (mounted) {
+                return;
+            }
+
+            setGotchisAmount(
+                guilds.map((guild, id) => response[id]?.ghosts_num || 0)
+            );
+        });
+
+        return () => mounted = true;
```
Comment (dudendy) [inline src/pages/Guilds/GuildsContext.js:51]: `mounted` means that component is initiated. 
I would suggest that we need a different approach without `return` statement inside promise (just if condition).

Same as we have in `BalancesContext.js`

File: src/pages/Guilds/components/GuildGotchis.js [hunk 3]
```diff
@@ -1,41 +1,60 @@
-import React, { useContext } from 'react';
+import { useContext, useEffect, useState } from 'react';
+import { CircularProgress } from '@mui/material';
 
 import Gotchi from 'components/Gotchi/Gotchi';
-import { GuildsContext } from 'contexts/GuildsContext';
+import GotchisLazy from 'components/Lazy/GotchisLazy';
+import thegraphApi from 'api/thegraph.api';
+import { GuildsContext } from '../GuildsContext';
```
Comment (dudendy) [inline src/pages/Guilds/components/GuildGotchis.js:7]: separate `GuildsContext` from current imports (see imports order)

File: src/pages/Guilds/components/GuildGotchis.js [hunk 4]
```diff
@@ -1,41 +1,60 @@
...
+
+        setIsGotchisLoading(true);
+
+        thegraphApi.getGotchisByAddresses(guilds[guildId].members).then(gotchis => {
+            if (mounted) {
+                setGuildGotchis(gotchis);
+            }
+        }).finally(() => setIsGotchisLoading(false));
+
+        return () => mounted = false;
```
Comment (dudendy) [inline src/pages/Guilds/components/GuildGotchis.js:33]: same thing with `mounted` what I described above

File: src/pages/Guilds/components/GuildLendings.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+        Promise.all(promises).then(responses => {
+            if (mounted) {
+                const lendings = responses.reduce((result, current) => result.concat(current), []);
+
+                setGuildLendings(lendings);
+            }
+        }).finally(() => setIsLendingsLoading(false));
+
+        return () => mounted = false;
```
Comment (dudendy) [inline src/pages/Guilds/components/GuildLendings.js:37]: "mounted" issue

File: src/pages/Guilds/routes/Guild.js [hunk 6]
```diff
@@ -1,122 +1,72 @@
...
-import GuildLogo from '../components/GuildLogo';
+import GuildBanner from '../components/GuildBanner';
+import GuildsDetails from '../components/GuildDetails';
 import GuildNav from '../components/GuildNav';
+import GuildGotchis from '../components/GuildGotchis';
+import GuildLendings from '../components/GuildLendings';
 import GuildsRealm from '../components/GuildsRealm';
-import { guildStyles } from '../styles';
+import { GuildsContext } from '../GuildsContext';
+import commonUtils from 'utils/commonUtils';
```
Comment (dudendy) [inline src/pages/Guilds/routes/Guild.js:14]: check imports order please

**Review by dudendy — CHANGES_REQUESTED** [review]

File: src/components/Modal/Modal.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import styles from './styles';
+import CloseIcon from '@mui/icons-material/Close';
+import { Box } from "@mui/system";
+import { IconButton, Modal } from "@mui/material";
+
+import classNames from 'classnames';
+
+export default function CustomModal({children, modalOpen, setModalOpen, className, handleModalClose}) {
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Modal/Modal.js]: fixed, thanks

File: src/components/Modal/Modal.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        };
+    }
+
+    return (
+        <Modal
+            open={modalOpen}
+            onClose={onModalClose}
+            className={classes.wrapper}
+        >
+            <Box className={classNames(classes.modal, className || null)} >
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Modal/Modal.js]: fixed, thanks

File: src/components/Modal/styles.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    modal: {
+        outline: 'none',
+        margin: 'auto',
+        position: 'relative',
+        backgroundColor: theme.palette.background.secondary,
+        borderRadius: 15,
+        padding: theme.spacing(2),
+        border: `1px solid ${theme.palette.secondary.light}`
+    },
+
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Modal/styles.js]: fixed, thanks

File: src/components/Modal/styles.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import { makeStyles } from "@mui/styles";
+
+const styles = makeStyles( theme => ({
+    wrapper: {
+        display: 'flex',
+        justifyContent: 'center',
+        overflow: 'auto',
+        padding: '10px'
+    },
+
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/components/Modal/styles.js]: fixed, thanks

File: src/pages/Guilds/GuildsContext.js (hunk 2 above)
Comment (dmutrokot) [reply src/pages/Guilds/GuildsContext.js]: fixed, thanks

File: src/pages/Guilds/components/GuildGotchis.js (hunk 3 above)
Comment (dmutrokot) [reply src/pages/Guilds/components/GuildGotchis.js]: fixed, thanks

File: src/pages/Guilds/components/GuildLendings.js (hunk 5 above)
Comment (dmutrokot) [reply src/pages/Guilds/components/GuildLendings.js:37]: seems here it works as designed

File: src/pages/Guilds/routes/Guild.js (hunk 6 above)
Comment (dmutrokot) [reply src/pages/Guilds/routes/Guild.js]: fixed, thanks

File: src/pages/Guilds/components/GuildGotchis.js (hunk 4 above)
Comment (dmutrokot) [reply src/pages/Guilds/components/GuildGotchis.js:33]: here works as designed

File: src/pages/Guilds/routes/GuildsPreview.js [hunk 11]
```diff
@@ -2,50 +2,52 @@ import React, { useContext, useEffect } from 'react';
...
 import { Box } from '@mui/system';
 import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import { Button, Tooltip } from '@mui/material';
 
-import { GuildsContext } from 'contexts/GuildsContext';
+import { FudIcon, GhstTokenIcon, GotchiIcon, LendingIcon, WarehouseIcon } from 'components/Icons/Icons';
+import GuildLogo from '../components/GuildLogo';
+import GuildWearables from '../components/GuildWearables';
+import { GuildsContext } from '../GuildsContext';
 import commonUtils from 'utils/commonUtils';
```
[the remark this answers is not in the document]
Comment (dmutrokot) [reply src/pages/Guilds/routes/GuildsPreview.js]: fixed, thanks

**Review by dmutrokot — APPROVED** [review]

**Review by dudendy — APPROVED** [review]

**Pull request merged**